web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1143,31 +1188,79 @@ namespace com.keyman.keyboards {
      *  @param  {string}   cmd        command string
      *  @param  {boolean?} byLanguage if true, context=languages, else context=keyboards
      **/
-    keymanCloudRequest(cmd: string, byLanguage?: boolean) {
-      var URL='https://api.keyman.com/cloud/4.0/', tFlag,
-        Lscript = this.keymanweb.util._CreateElement('script');
-
-      URL = URL + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards')
-        +'?jsonp=keyman.register&languageidtype=bcp47&version='+this.keymanweb['version'];
-
+    keymanCloudRequest(cmd: string, byLanguage?: boolean) {    
       var kbdManager = this;
+      var keymanweb = this.keymanweb;
+
+      // Some basic support toward #5044, but definitely not a full solution toward it.
+      // Wraps the cloud API keyboard-stub request in a Promise, allowing response on network
+      // and/or parser errors.  Also detects when `register` returns due to an error case that
+      // does not throw errors.  (There are a few such "empty" `return` statements there.)
+      const URL='https://api.keyman.com/cloud/4.0/'
+                + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards');
+
+      let promise = new Promise<void>(function(resolve, reject) {
+        const Lscript: HTMLScriptElement = keymanweb.util._CreateElement('script');
+
+        const queryConfig = '?jsonp=keyman.register&languageidtype=bcp47&version='+keymanweb['version'];
+  
+        // Set callback timer
+        const timeoutID = window.setTimeout(function() {
+          delete kbdManager.registrationResolvers[timeoutID];
+          reject(CLOUD_TIMEOUT_ERR);
+        } ,10000);
+
+        // Save the resolve / reject functions.
+        kbdManager.registrationResolvers[timeoutID] = {
+          resolve: resolve,
+          reject: reject
+        };
+
+        const tFlag='&timerid='+ timeoutID;
+
+        Lscript.onload = function(event: Event) {
+          // This case shouldn't happen, as `register` should handle all related
+          // Promise resolution / rejection... but just in case.
+          if(kbdManager.registrationResolvers[timeoutID]) {
+            delete kbdManager.registrationResolvers[timeoutID];
+            reject(CLOUD_STUB_REGISTRATION_ERR);
```

Comment:
Here you do `delete` then `reject`, but on line 1235-1239 you do it the other way around.